dvc/repo/experiments/executor/base.py

```diff
@@ -348,6 +348,7 @@ def filter_pipeline(stages):
                     quiet=True,
                     allow_missing=True,
                     checkpoint_reset=checkpoint_reset,
+                    recursive=kwargs.get("recursive", False),
```

Comment:
@pmrowla, Is it not possible to checkout as we go `repro`-ing? This way, it's may miss a lots of things. `foreach` based stages come to mind, as only `repro` supports them.

Reply:
Yeah we could do that. I think we'd just need to pass a flag into repro (and subsequently `stage.run`) that indicates we need to `checkout(allow_missing=True)` dependencies prior to reproducing a stage, right?

Reply:
The one thing I'm not sure on is how this affects the run cache lookup - do deps need to be checked out before that lookup or can we just skip the dep checkout entirely in the event we have a cached stage?

Reply:
Run-cache lookup does get affected, as if there's no dependency, there's no cached stage. This will work for workspace based run, but we may need to map dependencies into the tmpdir repo.